app/services/SubmissionService.scala

```diff
@@ -0,0 +1,151 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package services
+
+import java.time.LocalDate
+import java.time.temporal.ChronoUnit
+
+import config.FrontendAppConfig
+import connectors.{CitizenDetailsConnector, TaiConnector}
+import javax.inject.{Inject, Singleton}
+import models.auditing.AuditEventType._
+import models.requests.DataRequest
+import models.{AuditData, FlatRateItem}
+import play.api.Logger
+import play.api.mvc.AnyContent
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.play.audit.http.connector.AuditConnector
+import utils.TaxYearDates._
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.language.postfixOps
+
+
+@Singleton
+class SubmissionService @Inject()
+(
+  citizenDetailsConnector:  CitizenDetailsConnector,
+  taiConnector:             TaiConnector,
+  auditConnector:           AuditConnector,
+  appConfig:                FrontendAppConfig
+) {
+
+  val ONE_WEEK  = 1
+  val ZERO      = 0
+
+
+  def submitExpenses(startDate: LocalDate)
+                    (implicit dataRequest: DataRequest[AnyContent], hc: HeaderCarrier, ec: ExecutionContext): Future[Either[String, Unit]] = {
+
+    submit(startDate) map {
+
+      case Right(submittedDetails) =>
+        Logger.info(s"[SubmissionService][submitExpenses] Submission successful")
+        auditSubmissionSuccess(submittedDetails)
+        Right(())
+
+      case Left(error) =>
+        Logger.error(s"[SubmissionService][submitExpenses] Submission failed : $error")
+        auditSubmissionFailure(error)
+        Left(error)
+    }
+  }
+
+
+  private def submit(startDate: LocalDate)
+            (implicit dataRequest: DataRequest[AnyContent], hc: HeaderCarrier, ec: ExecutionContext): Future[Either[String, Seq[FlatRateItem]]] = {
+
+    val flatRateItems: Seq[FlatRateItem] = Seq[FlatRateItem](
+      FlatRateItem(year = TAX_YEAR_2019_START_DATE.getYear, amount = calculate2019FlatRate(startDate)),
+      FlatRateItem(year = TAX_YEAR_2020_START_DATE.getYear, amount = calculate2020FlatRate())
+    )
+
+    def futureSequence[I, O](inputs: Seq[I])(flatMapFunction: I => Future[O])(implicit ec: ExecutionContext): Future[Seq[O]] =
+      inputs.foldLeft(Future.successful(Seq.empty[O]))(
+        (previousFutureResult, nextInput) =>
+          for {
+            futureSeq <- previousFutureResult
+            future    <- flatMapFunction(nextInput)
+          } yield futureSeq :+ future
+      )
+
+    futureSequence(flatRateItems) {
+      item: FlatRateItem =>
+        for {
+          etag <- citizenDetailsConnector.getETag(dataRequest.nino)
+          _    <- taiConnector.postIabdData(dataRequest.nino, item.year, item.amount, etag)
+        } yield item
+    } map {
+      submittedDetails => Right(submittedDetails)
+    } recover {
+      case e => Left(e.getMessage)
+    }
+  }
+
+
+  def numberOfWeeks(startDate:LocalDate, endDate:LocalDate): Long = {
```

Comment:
val startDate = LocalDate.of(2020, 3, 1)
serviceUnderTest.numberOfWeeks(startDate, LocalDate.of(2020, 3, 16)) shouldBe 3
It looks the it is always rounding up which is good. This should have been 2 otherwise.

Reply:
yep, we had confirmation that we have to round up to the next whole week